### ISSUE TITLE
AutoFocus does not work on an input

### ISSUE THREAD

**Issue by pbrzosko** [ant-design/ant-design#49937]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/antd-reproduce-5x-8zjer3?file=demo.tsx)

### Steps to reproduce

Click "Click me" button.

### What is expected?

Popover is displayed and input field has focus.

### What is actually happening?

Input does not have focus. Autofocus property is not even rendered on the HTML input element.

| Environment | Info |
| --- | --- |
| antd | 5.18.2 |
| React | 18.3.1 |
| System | macOS Sonoma 14.5 |
| Browser | Google Chrome Version 126.0.6478.127 (Official Build) (arm64) |

**Comment by nova1751** [comment]
React has special handling for the global attribute `autofocus `. See the two posts below: 
https://blog.maisie.ink/react-ref-autofocus/
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus
If you want to make the input element focused when opening the popover, check the example blow:
https://stackblitz.com/edit/antd-reproduce-5x-npjsfv?file=demo.tsx
Hope this can help you.😊

**Comment by pbrzosko** [comment]
I am just migrating project from Antd4 to Antd5 and I just noticed, that before we had autofocus attribute and it all worked. After migrating to Antd5, this attribute is not even rendered on the input itself, so it will definitely not work. And this si something I think should be fixed.
I tried to call focus method manually on the input after component is mounted (we have an input wrapper), but this didn't work. I will try different approaches like you pasted, so that it is focused when popover is opened.

But anyway, I think autofocus attribute should be propagated to the input.

**Comment by nova1751** [comment]
> I am just migrating project from Antd4 to Antd5 and I just noticed, that before we had autofocus attribute and it all worked. After migrating to Antd5, this attribute is not even rendered on the input itself, so it will definitely not work. And this si something I think should be fixed. I tried to call focus method manually on the input after component is mounted (we have an input wrapper), but this didn't work. I will try different approaches like you pasted, so that it is focused when popover is opened.
> 
> But anyway, I think autofocus attribute should be propagated to the input.

I thought the same as you previously, lol. However, React handles the `autoFocus` attribute specially, which makes the attribute not be transfered to dom element. You can try to edit the demo or read the first post I pasted above to verify that.

**Comment by pbrzosko** [comment]
Alright, I need to check what is rendered for Antd4 vs Antd5. Because I just assumed that autofocus attribute is missing but maybe the reason for test to fail is titally differrent. I will investigate that. Thanks for the links.

**Comment by pbrzosko** [comment]
Ok, I have checked that it was the same behavior in Antd4. But I guess popover was rendered differently, and seems like autofocus worked on an input inside a popover component.
Now I had to call focus manually, after input is mounted and also with a setTimeout(0), but it works.

Thus, I am closing the issue.
